connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -73,6 +73,11 @@ class SparkSession private[sql] (
     with Logging {
 
   private[this] val allocator = new RootAllocator()
+  private var shouldStopCleaner = false
+  private[sql] lazy val cleaner = {
+    shouldStopCleaner = true
+    new ContextCleaner(this)
```

Comment:
Why not use the JVM one?

Reply:
The problem is that the interface forces to fix generated code `org.apache.spark.connect.proto.CachedRemoteRelation` to inherit `AutoCloseable`, or I have to create a wrapper (see https://docs.oracle.com/javase/9/docs/api/java/lang/ref/Cleaner.html)

Reply:
And seems like Java protobuf gen does not allow it:
> Don’t go looking for facilities similar to class inheritance, though – protocol buffers don’t do that.

https://protobuf.dev/getting-started/javatutorial/#defining-your-protocol-format

Reply:
Let me make it simpler at least.